mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -917,9 +895,32 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+    private void restartAfterSettingsUpdate() {
+        restartAfterSettingsUpdate(false);
+    }
+
+    private void restartAfterSettingsUpdate(boolean firstTimeDemo) {
+        // Restart app after preferences
+        Log.d(TAG, "Restarting after settings");
+        // Get launch intent for application
+        Intent restartIntent = getBaseContext().getPackageManager()
```

Comment:
Do you need `getBaseContext()` here?

Reply:
Nope. You need getBaseContext() almost never, except for some theme related edge cases (e.g. when you have a themed action bar context and want to resolve colors against the activity context).

Reply:
This code is simply copied, I'll change it :)